mscore/menus.cpp

```diff
@@ -1526,7 +1526,13 @@ Palette* MuseScore::newFretboardDiagramPalette()
 
 void MuseScore::setAdvancedPalette()
       {
-      mscore->getPaletteBox();
```

Comment:
I'm not clear on what is going on with this change (or the similar one in setBasicPalette().  Can you describe?

Reply:
I'm getting the shortcuts from the palette cells, and then setting those shortcuts in the new palette cells. setAdvancedPalette() and setBasicPalette() get called each time the palettes are redrawn, so the shortcuts need to be preserved by saving them before the palette box is cleared, and then setting the shortcuts in the new palette box.

Reply:
OK, I believe that makes sense.  Previously, there was no state that needed preserving - we could just rebuild these from scratch each time we needed to.  Now, we need to at least remember the shortcuts assigned.  I guess I was thinking somehow we might be reading/writing these on switch if there had been any customization, and the shortcuts would take care of themselves.  Would that also have been a valid approach?  Any advantages/disadvantages?

Reply:
I think this would also be valid, but many changes would need to be made to use that approach. There were some problems if the workspace was read each time (palette shortcuts would stop working when the workspace was toggled), so to fix this, the palettes are read only on startup, and after that only the shortcuts are stored. Shortcuts are saved, stored each time the workspace is switched.

Reply:
OK, thanks for the feedback.  I was wondering if the workspace save/restore would be easier, but if not, just keep it the way it is.